FV3/atmos_cubed_sphere/tools/fv_mp_mod.F90

```diff
@@ -635,14 +643,49 @@ subroutine domain_decomp(npx,npy,nregions,grid_type,nested,Atm,layout,io_layout)
                   tile_id(n) = n
                enddo
             endif
+            allocate(xextent(layout(1),nregions), yextent(layout(2),nregions))
```

Comment:
Maybe isolate this logic to a new subroutine?

Reply:
Done.